ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py

```diff
@@ -0,0 +1,306 @@
+            self.source_config.schemaFilterPattern, schema_name
+        ):
+            self.status.filter(
+                schema_name, f"Schema pattern not allowed for schema {schema_name}"
+            )
+            return True
+
+        return False
+
+    def _filter_tables(self, table: Table) -> bool:
+        """Filter tables based on the table filter pattern
+
+        Args:
+            tables (Iterable[Table]):
+
+        Returns:
+            Iterable[Table]:
+        """
+        table_name = table.name.root
+        if table.fullyQualifiedName and self.source_config.useFqnForFiltering:
+            table_name = table.fullyQualifiedName.root
+
+        if filter_by_table(self.source_config.tableFilterPattern, table_name):
+            self.status.filter(
+                table_name, f"Table pattern not allowed for table {table_name}"
+            )
+            return True
+
+        return False
+
+    def _filter_column_metrics_computation(self):
+        """Filter"""
+
+    def _get_database_entities(self) -> Iterable[Database]:
+        """Get database entities"""
+        if not self.config.source.serviceName:
+            raise ValueError("serviceName must be provided in the source configuration")
+
+        databases = self.metadata.list_all_entities(
+            entity=Database,
+            params={"service": self.config.source.serviceName},
+        )
+        databases = cast(Iterable[Database], databases)
+
+        if self.source_config.databaseFilterPattern:
+            databases = self._filter_databases(databases)
+
+        if not databases:
+            raise ValueError(
+                "databaseFilterPattern returned 0 result. At least 1 database must be returned by the filter pattern."
+                f"\n\t- includes: {self.source_config.databaseFilterPattern.includes if self.source_config.databaseFilterPattern else None}"  # pylint: disable=line-too-long
+                f"\n\t- excludes: {self.source_config.databaseFilterPattern.excludes if self.source_config.databaseFilterPattern else None}"  # pylint: disable=line-too-long
+            )
+
+        return cast(Iterable[Database], databases)
+
+    def _filter_entities(self, tables: Iterable[Table]) -> Iterable[Table]:
+        """Filter entities based on the filter pattern
+
+        Args:
+            entities (Iterable[EntityInterfaceWithTags]): Entities to filter
+
+        Returns:
+            Iterable[EntityInterfaceWithTags]
+        """
+        tables = [
+            table
+            for table in tables
+            if (
+                not self.source_config.schemaFilterPattern
+                or not self._filter_schemas(table)
+            )
+            and (
+                not self.source_config.tableFilterPattern
+                or not self._filter_tables(table)
+            )
+        ]
+
+        if self.source_config.classificationFilterPattern:
+            tables = self.filter_classifications(tables)  # type: ignore
```

Comment:
I see the `filter_classifications` eats an `Iterable`. Would it make sense to have the `_filter_*` methods to work on individual assets? Might make it easier to reuse and chain filters in a single comprehension, something like:

```
[
            table
            for table in tables
            if (
                not self.source_config.schemaFilterPattern
                or not self._filter_schemas(table)
            )
            and (
                not self.source_config.tableFilterPattern
                or not self._filter_tables(table)
            )
            and  not self._filter_classification(table)
        ]
```

Reply:
Yep, that makes sense. I started with the iterable and then made schema and table eat an entity. Let me update that.